crates/evm/traces/src/lib.rs

```diff
@@ -81,10 +84,10 @@ const RETURN: &str = "← ";
 /// Render a collection of call traces.
 ///
 /// The traces will be decoded using the given decoder, if possible.
-pub async fn render_trace_arena(
+pub async fn render_trace_arena<'a>(
     arena: &CallTraceArena,
     decoder: &CallTraceDecoder,
-) -> Result<String, std::fmt::Error> {
+) -> Result<(String, Vec<String>), std::fmt::Error> {
```

Comment:
I am adding the logic in this module itself to prevent code duplication, since logic for the folded stack trace is minimal here. Maybe the function `render_trace_arena` be renamed or some refactoring.